modules/external-api/src/main/java/org/opencastproject/external/common/ApiVersion.java

```diff
@@ -20,30 +20,63 @@
  */
 package org.opencastproject.external.common;
 
+import static org.opencastproject.external.common.ApiMediaType.VERSION_REG_EX_PATTERN;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 /** The existing versions */
 public enum ApiVersion {
 
   VERSION_UNDEFINED("v*.*.*"),
 
-  VERSION_1_0_0("v1.0.0");
+  VERSION_1_0_0("v1.0.0"),
+
+  VERSION_1_1_0("v1.1.0");
 
-  public static final ApiVersion CURRENT_VERSION = VERSION_1_0_0;
+  public static final ApiVersion CURRENT_VERSION = VERSION_1_1_0;
 
   private String versionString;
+  private int major;
+  private int minor;
+  private int micro;
 
   ApiVersion(String versionString) {
     this.versionString = versionString;
+    if ("v*.*.*".equals(versionString)) {
+      major = Integer.MAX_VALUE;
+      minor = Integer.MAX_VALUE;
+      micro = Integer.MAX_VALUE;
+    } else {
+      Matcher matcher = Pattern.compile(VERSION_REG_EX_PATTERN).matcher(versionString);
+      if (!matcher.matches() || matcher.groupCount() != 3) {
+        throw new IllegalArgumentException("'" + versionString + "' is not a valid version");
+      }
+      major = Integer.parseInt(matcher.group(1));
+      minor = Integer.parseInt(matcher.group(2));
+      micro = Integer.parseInt(matcher.group(3));
+    }
   }
 
   public static ApiVersion of(String version) {
     switch (version) {
       case "v1.0.0":
         return VERSION_1_0_0;
+      case "v1.1.0":
+        return VERSION_1_1_0;
       default:
         throw new IllegalArgumentException("'" + version + "' is not a valid version");
     }
   }
 
+  public boolean isSmallerThan(ApiVersion other) {
+    if (this.major < other.major) return true;
```

Comment:
Personally, I would prefer...

```
return ((this.major < other.major) ||
             ((this.major == other.major) && (this.minor < other.minor)) ||
             ((this.major == other.major) && (this.minor == other.minor)&& (this.patch < other.patch));
```

... but this might be a matter of taste.